src/plots/cartesian/layout_defaults.js

```diff
@@ -55,22 +63,47 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         }
     }
 
-    function axSort(a,b) {
-        var aNum = Number(a.substr(5)||1),
-            bNum = Number(b.substr(5)||1);
-        return aNum - bNum;
+    // N.B. Ignore orphan axes (i.e. axes that have no data attached to them)
+    // if gl3d or geo is present on graph. This is retain backward compatible.
+    //
+    // TODO drop this in version 2.0
+    var ignoreOrphan = (layoutOut._hasGL3D || layoutOut._hasGeo);
```

Comment:
> an we confirm that we're not generating this situation anymore?

`scatter3d` traces are still generated with a `layout.xaxis` and `layout.yaxis` in the workspace.